pkg/system/gateway/announce.go

```diff
@@ -127,7 +128,7 @@ func (a *announcer) announceRemoteNode(ctx context.Context) {
 	var deviceChanges <-chan rx.Change[remoteDesc]
 	var stopDevicesSub context.CancelFunc
 	// So we can undo the API proxy and metadata announcement separately.
-	var undoDevices, undoMD tasks
+	var undoDevices, undoMD, removeChecks tasks
```

Comment:
I can't find when the `removeChecks` functions are actually invoked. I see where they are added / removed